pkg/agent/openflow/pipeline.go

```diff
@@ -1259,6 +1259,7 @@ func (c *client) l3FwdFlowToRemoteViaGW(
 	return l3FwdTable.BuildFlow(priorityNormal).MatchProtocol(ipProto).
 		MatchDstIPNet(peerSubnet).
 		Action().SetDstMAC(localGatewayMAC).
+		Action().SetSrcMAC(c.nodeConfig.UplinkNetConfig.MAC).
```

Comment:
Note this function is also used by normal noEncap case, setting src MAC unconditionally would lead to some unexpected behavior. For example, the variable should be nil in linux noEncap case.

Perhaps we should create a separate method to differentiate local gateway and remote gateway.

Reply:
Thanks. I moved this to a dedicated function to avoid potential issues.